util.go

```diff
@@ -240,15 +242,21 @@ func asIntFromTimeDuration(param string) int64 {
 	return int64(d)
 }
 
+// GlobalTimeDurationTypes holds specific application registered custom
+// time.Duration types so their values can be properly parsed as time.Duration.
+var GlobalTimeDurationTypes = []reflect.Type{timeDurationType}
+
 // asIntFromType calls the proper function to parse param as int64,
 // given a field's Type t.
 func asIntFromType(t reflect.Type, param string) int64 {
-	switch t {
-	case timeDurationType:
+	for _, customDurationType := range GlobalTimeDurationTypes {
+		if t != customDurationType {
+			continue
+		}
 		return asIntFromTimeDuration(param)
```

Comment:
I'd reverse this logic:
```suggestion
		if t == customDurationType {
			return asIntFromTimeDuration(param)
		}
```